app/models/concerns/orchestration/dns.rb

```diff
@@ -18,6 +19,20 @@ def reverse_dns?
     (host.nil? || host.managed?) && managed? && hostname.present? && ip_available? && !subnet.nil? && subnet.dns? && SETTINGS[:unattended]
   end
 
+  def rebuild_dns
+    begin
+      if dns? || reverse_dns?
+        recreate_a_record && recreate_ptr_record
```

Comment:
You should only call the recreate_X_record if that type of DNS is enabled, so don't call recreate_ptr_record is dns? is true and reverse_dns? is false.

I think using `&&` here could be a problem too, as it would skip rebuild_ptr_record if rebuild_a_record returned a falsey value.